### DISCUSSION TITLE
Jackbox Mouse support

### DISCUSSION THREAD

**Discussion by AkiraArtuhaxis** [AlexisL61/JackboxPatcherFR#257] [Ideas]
Usually, when I'm playing with my friends locally, I have a laptop set up to a TV and I control everything using a mouse. However, this way you can't really pause the games.
That's why I made [this script](https://github.com/AlexisL61/JackboxUtility/files/11996459/JackboxMouse.zip) (with AutoHotKey) in which you can control everything you need from Jackbox using only your mouse.
The controls are on the image there, but the basic is Right Click to Pause and Middle Click to close the game.
I suppose the implementation into the Utility wouldn't be difficult. It'd be just opening and closing the script as long as the Utility is open and not letting the script interact with the Utility (so you don't accidentally close the Utility with the middle click.)
![JackboxMouseMockup](https://github.com/AlexisL61/JackboxUtility/assets/93792993/cd84dfeb-ebde-4966-b776-e6ce335dede9)

**Comment by MeDustyy** [comment]
This makes no sense to 99% of app users. Why does it have to be in the app? Why can't the 1% just use AutoHotKey or software from Razer, Logitech, and other mouse manufacturers?

**Comment by AkiraArtuhaxis** [comment]
There are various reasons as to why this has to be in the app:

1. With the usual input software you mention, as far as I'm concerned you can't make it so the input program doesn't interact with a specific program (in this case, the Utility). By being directly implemented into the program, accidentally closing the Utility with the middle click would be prevented.
2. By being directly in the app, the input software would automatically start and finish as long as you have the Utility open.
3. Normal users don't know how to do this kind of stuff. It actually took me quite a bit of research and testing to do this properly. So simplifying it to a toggle would be pretty nice.

Right now, there are only 3 Settings sub-menus, so one more really wouldn't hurt. This is no doubt my most requested feature, as it's extremely good when you're playing in local sessions. Please do not say it's not useful to 99% of users, as you don't know how many users play locally instead of by streaming the games (I'm one of those btw).

**Comment by MisterAlexEsquire** [comment]
It looks like AHK thinks that the linked script requires a deprecated version of AHK. Does it need to be edited to specify the current version (or that it is not version-specific)?

![image](https://github.com/user-attachments/assets/ca1d5159-9451-46ba-8465-09256c73f031)

**Reply by AkiraArtuhaxis** [reply]
That script is kinda outdated now. The most recent version I made is an exe which, if put in the same folder as the Utility updater, will open and close at the same time as the Utility. The script will also only affect the Jackbox Utility games and nothing else. You can get it here: [JackboxMouse.zip](https://github.com/user-attachments/files/16592631/JackboxMouse.zip)

**Comment by AkiraArtuhaxis** [comment]
This, alongside a few more things, is now part of the [Jackbox Utility Extras](https://github.com/JackboxUtility/Jackbox-Utility-Extras).
